### ISSUE TITLE
Why reduction of inflowVelocity_ by spanwiseVelocity_ before calculating the relativeVelocity_?

### ISSUE THREAD

**Issue by samfredriksson** [turbinesFoam/turbinesFoam#255]
In line 660 in actuatorLineElement.C there is a reduction of inflowVelocity_ by spanwiseVelocity_ as 

    inflowVelocity_ -= spanwiseVelocity;

This is done before calculating the relativeVelocity_ which I have some problems with. I reason that the reduction is done in order to have a normal velocity towards the wing as it is used in the force calculations using Cl and Cd.  However, since the calculation of the relativeVelocity_ is performed later there might once again be a velocity component in the spanwise direction or am I missing something here. Please elaborate.

//Sam

**Comment by petebachant** [comment]
As you said, this is done to remain consistent with how 2-D airfoil coefficients are measured, i.e., they use the wind tunnel free stream velocity. There would only be a spanwise component in `relativeVelocity` if the element was moving along its span, which shouldn't be true.

**Comment by samfredriksson** [comment]
Seems reasonable but I have added a possibility to use an arbitrary path of the ALM which then can create velocities along the span and then need to do the adjustment later in the process. I just wanted to be sure that I did not miss anything.
Thanks for the reply :-)

**Comment by petebachant** [comment]
Sounds good. @andersgoude actually added the ability to sample along a circle around the element. Feel free to open up a PR and mark it WIP if you want some early code review.

**Comment by andersgoude** [comment]
This phenomena is not directly related to the velocity sampling. The main idea for removing the spanwise component can be seen if one instead assumes that we have a constant flow direction (let's say the x-direction) and we have an airfoil with the cross-section in the x-y plane. Now, if you make the airfoil itself move in the z-direction, this should leave the cross-section in the x-y plane constant, and if we neglect 3D boundary layer effects, the flow pattern should therefore also remain constant. If the flow is constant, the forces on the blade has to be constant as well. Therefore, the blade movement in the z-direction should not affect the forces. Now, if the blade movement does not affect the forces, the same should be true if the blade is stationary, but instead we add a flow component in the z-direction. This is the main reason why one removes the spanwise velocity component before calculating the relative velocities.

**Comment by andersgoude** [comment]
I don't think the sampling of the velocity in a circle will be affected by any of these changes.

To make the code more general, I guess we could project the full relative velocity on the normal and tangential directions, instead of subtracting the spanwise component from the inflow.

It looks like the current code can get issues with the calculation of relativeVelocityGeom_ as well, as the spanwise component is not subtracted here.

**Comment by samfredriksson** [comment]
I agree that this does not affect the sampling and I think that the code would be more general by changing 

    // Subtract spanwise component of inflow velocity
    vector spanwiseVelocity = spanDirection_
                            * (inflowVelocity_ & spanDirection_)
                            / magSqr(spanDirection_);
    inflowVelocity_ -= spanwiseVelocity;

    // Calculate relative velocity and Reynolds number
    relativeVelocity_ = inflowVelocity_ - velocity_;
    Re_ = mag(relativeVelocity_)*chordLength_/nu_;_

with something like this

    // Calculate relative velocity and Reynolds number
    relativeVelocity_ = inflowVelocity_ - velocity_;

    // Subtract spanwise component of inflow velocity
    vector spanwiseVelocity = spanDirection_
                            * (relativeVelocity_ & spanDirection_)
                            / magSqr(spanDirection_);
    relativeVelocity_ -= spanwiseVelocity;
    Re_ = mag(relativeVelocity_)*chordLength_/nu_;

And yes it seems to affect relativeVelocityGeom_ as well which, however, is of less importance for me since it is only for post processing purposes?

**Comment by andersgoude** [comment]
That looks correct to me
